rcgen/src/lib.rs

```diff
@@ -156,6 +156,29 @@ pub enum SanType {
 	DnsName(Ia5String),
 	URI(Ia5String),
 	IpAddress(IpAddr),
+	OtherName((Vec<u64>, OtherNameValue)),
+}
+
+/// An OtherName value, defined in [RFC 5280§4.1.2.4].
+///
+/// Technically, this could be any ASN.1 types but for now we cover only UTF-8 strings
+/// as it will cover most of use cases, for instance smart cards.
+///
+/// [RFC 5280§4.1.2.4]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+#[non_exhaustive]
+pub enum OtherNameValue {
```

Comment:
The RFC calls this `AnotherName`. Maybe we should use that instead?

Reply:
Actually, they use both. In the main RFC ( [RFC 5280§4.2.1.6](https://datatracker.ietf.org/doc/html/rfc5280#page-38)) they use `OtherName` and in the [appendix](https://datatracker.ietf.org/doc/html/rfc5280#page-128) they use `AnotherName`, they also justify this in the appendix:

> -- AnotherName replaces OTHER-NAME ::= TYPE-IDENTIFIER, as
-- TYPE-IDENTIFIER is not supported in the '88 ASN.1 syntax

In [OpenSSL](https://www.openssl.org/docs/man1.0.2/man5/x509v3_config.html) they use `OtherName`, in our dependencies (for instance [x509-parser](https://docs.rs/x509-parser/latest/x509_parser/extensions/enum.GeneralName.html#variant.OtherName)), they also use it. I think we should stick with `OtherName`.

Reply:
Fair points! :+1: